### ISSUE TITLE
Unqualified service names do not resolve when using --docker-run with v2.8.5, last working v2.7.6 

### ISSUE THREAD

**Issue by craftey** [telepresenceio/telepresence#2870]
Unqualified services names do not resolve when using the --docker-run flag of the telepresence intercept command. From the outside it looks like a regression of fixed #2777.

I reproduced it on GCP with telepresence v2.8.5. as follows:

Create k8s-cluster and deployment in GCP according to this howto:
https://cloud.google.com/kubernetes-engine/docs/deploy-app-cluster#standard
You can adjust the zone to your likings.

```
gcloud container clusters create k8s-test-cluster --num-nodes=1 --zone=europe-west3-a
gcloud container clusters get-credentials k8s-test-cluster --zone=europe-west3-a
kubectl create namespace test

for SERVER in hello-server hello-server2 ; do
    kubectl create deployment $SERVER --namespace test --image=us-docker.pkg.dev/google-samples/containers/gke/hello-app:1.0
    kubectl rollout status deployment/$SERVER --namespace test
    kubectl expose deployment $SERVER --type LoadBalancer --port 80 --target-port 8080 --namespace test
done
```
Then run the following commands:
```
telepresence helm install

helm list -n ambassador 
#NAME           	NAMESPACE 	REVISION	UPDATED                              	STATUS  	CHART             	APP VERSION
#traffic-manager	ambassador	1       	2022-09-29 13:15:41.374646 +0200 CEST	deployed	telepresence-2.7.6	2.7.6      

telepresence intercept hello-server --namespace test \
  --docker-run -- --rm curlimages/curl:7.85.0 \
  -v --connect-timeout 10 \
  hello-server2:80
```
Fails with `curl: (6) Could not resolve host: hello-server2`
daemon.log:
```
2022-11-07 17:20:44.0369 info    daemon/session/dns/SearchPaths : setting search paths ambassador default kube-node-lease kube-public kube-system test test.svc.cluster.local. tel2-search
2022-11-07 17:20:44.0373 debug   daemon/session/watch-cluster-info : Opening tunnel for id tcp 10.60.0.0:54684 -> 10.60.0.40:33335
2022-11-07 17:20:44.0374 info    daemon/session/dns/SearchPaths : Generated new /etc/resolver/telepresence.tel2-search.local
2022-11-07 17:20:44.2529 debug   daemon/session/dns/Server : ServeDNS 28150 AAAA   hello-server.test.svc.cluster.local.
2022-11-07 17:20:44.2531 debug   daemon/session/dns/Server : Lookup AAAA "hello-server.test.svc.cluster.local."
2022-11-07 17:20:44.2990 debug   daemon/session/dns/Server : 28150 AAAA   hello-server.test.svc.cluster.local. -> NOERROR EMPTY
2022-11-07 17:20:44.3011 debug   daemon/session/dns/Server : ServeDNS 65299 AAAA   hello-server.cluster.local.
2022-11-07 17:20:44.3012 debug   daemon/session/dns/Server : 65299 AAAA   hello-server.cluster.local. -> NXDOMAIN 
2022-11-07 17:20:44.3673 debug   daemon/session/dns/Server : ServeDNS 30457 A      hello-server.test.svc.cluster.local.
2022-11-07 17:20:44.3675 debug   daemon/session/dns/Server : Lookup A "hello-server.test.svc.cluster.local."
2022-11-07 17:20:44.4359 debug   daemon/session/dns/Server : 30457 A      hello-server.test.svc.cluster.local. -> NOERROR [10.64.9.110]
2022-11-07 17:20:44.9159 debug   daemon/session/dns/Server : ServeDNS 55858 A      hello-server2.tel2-search.
2022-11-07 17:20:44.9160 debug   daemon/session/dns/Server : Lookup A "hello-server2."
2022-11-07 17:20:44.9159 debug   daemon/session/dns/Server : ServeDNS 33988 AAAA   hello-server2.tel2-search.
2022-11-07 17:20:44.9162 debug   daemon/session/dns/Server : Lookup AAAA "hello-server2."
2022-11-07 17:20:44.9537 debug   daemon/session/dns/Server : 33988 AAAA   hello-server2. -> NOERROR EMPTY
2022-11-07 17:20:44.9537 debug   daemon/session/dns/Server : 55858 A      hello-server2. -> NOERROR [10.64.2.211]
2022-11-07 17:20:45.9658 debug   daemon/session/dns/Server : ServeDNS 33988 AAAA   hello-server2.tel2-search.
2022-11-07 17:20:45.9658 debug   daemon/session/dns/Server : ServeDNS 55858 A      hello-server2.tel2-search.
2022-11-07 17:20:45.9661 debug   daemon/session/dns/Server : 33988 AAAA   hello-server2.tel2-search. -> NOERROR EMPTY
2022-11-07 17:20:45.9662 debug   daemon/session/dns/Server : 55858 A      hello-server2.tel2-search. -> NOERROR [10.64.2.211]
2022-11-07 17:20:45.9698 debug   daemon/session/dns/Server : ServeDNS 64203 SOA    hello-server2.tel2-search.
2022-11-07 17:20:45.9700 debug   daemon/session/dns/Server : Lookup SOA "hello-server2."
2022-11-07 17:20:47.0011 debug   daemon/session/dns/Server : ServeDNS 64203 SOA    hello-server2.tel2-search.
2022-11-07 17:20:47.0080 debug   daemon/session/dns/Server : 64203 SOA    hello-server2.tel2-search. -> NXDOMAIN 
2022-11-07 17:20:47.0080 debug   daemon/session/dns/Server : 64203 SOA    hello-server2. -> SERVFAIL rpc error: code = Unimplemented desc = unsupported DNS query type SOA
2022-11-07 17:20:47.2371 debug   daemon/session/dns/SearchPaths : [ambassador default kube-node-lease kube-public kube-system test test.svc.cluster.local. tel2-search] -> [ambassador default kube-node-lease kube-public kube-system test]
2022-11-07 17:20:47.2371 info    daemon/session/dns/SearchPaths : setting search paths ambassador default kube-node-lease kube-public kube-system test
2022-11-07 17:20:47.2372 info    daemon/session/dns/SearchPaths : Removing /etc/resolver/telepresence.tel2-search.local
```

Curl without docker works:
```
telepresence intercept hello-server --namespace test \
  -- curl \
  -v --connect-timeout 10 \
  hello-server2:80
```
daemon.log:
```
2022-11-07 17:23:59.7423 info    daemon/session/dns/SearchPaths : setting search paths ambassador default kube-node-lease kube-public kube-system test test.svc.cluster.local. tel2-search
2022-11-07 17:23:59.7424 debug   daemon/session/watch-cluster-info : Opening tunnel for id tcp 10.60.0.0:54708 -> 10.60.0.40:33335
2022-11-07 17:23:59.7433 info    daemon/session/dns/SearchPaths : Generated new /etc/resolver/telepresence.tel2-search.local
2022-11-07 17:23:59.9590 debug   daemon/session/dns/Server : ServeDNS 30656 AAAA   hello-server.test.svc.cluster.local.
2022-11-07 17:23:59.9591 debug   daemon/session/dns/Server : Lookup AAAA "hello-server.test.svc.cluster.local."
2022-11-07 17:24:00.0005 debug   daemon/session/dns/Server : 30656 AAAA   hello-server.test.svc.cluster.local. -> NOERROR EMPTY
2022-11-07 17:24:00.0050 debug   daemon/session/dns/Server : ServeDNS 23218 AAAA   hello-server.cluster.local.
2022-11-07 17:24:00.0056 debug   daemon/session/dns/Server : 23218 AAAA   hello-server.cluster.local. -> NXDOMAIN 
2022-11-07 17:24:00.0906 debug   daemon/session/dns/Server : ServeDNS 27482 A      hello-server.test.svc.cluster.local.
2022-11-07 17:24:00.0909 debug   daemon/session/dns/Server : Lookup A "hello-server.test.svc.cluster.local."
2022-11-07 17:24:00.1419 debug   daemon/session/dns/Server : 27482 A      hello-server.test.svc.cluster.local. -> NOERROR [10.64.9.110]
2022-11-07 17:24:00.2673 debug   daemon/session/dns/Server : ServeDNS 51232 A      hello-server2.test.svc.cluster.local.
2022-11-07 17:24:00.2674 debug   daemon/session/dns/Server : Lookup A "hello-server2.test.svc.cluster.local."
2022-11-07 17:24:00.2689 debug   daemon/session/dns/Server : ServeDNS 41969 AAAA   hello-server2.test.svc.cluster.local.
2022-11-07 17:24:00.2689 debug   daemon/session/dns/Server : Lookup AAAA "hello-server2.test.svc.cluster.local."
2022-11-07 17:24:00.3025 debug   daemon/session/dns/Server : 51232 A      hello-server2.test.svc.cluster.local. -> NOERROR [10.64.2.211]
2022-11-07 17:24:00.3071 debug   daemon/session/dns/Server : 41969 AAAA   hello-server2.test.svc.cluster.local. -> NOERROR EMPTY
2022-11-07 17:24:00.3083 debug   daemon/session/dns/Server : ServeDNS  4256 AAAA   hello-server2.cluster.local.
2022-11-07 17:24:00.3084 debug   daemon/session/dns/Server :  4256 AAAA   hello-server2.cluster.local. -> NXDOMAIN 
2022-11-07 17:24:00.3104 debug   daemon/session/watch-cluster-info : Opening tunnel for id tcp 10.64.0.0:54711 -> 10.64.2.211:80
2022-11-07 17:24:00.4198 debug   daemon/session/dns/SearchPaths : [ambassador default kube-node-lease kube-public kube-system test test.svc.cluster.local. tel2-search] -> [ambassador default kube-node-lease kube-public kube-system test]
2022-11-07 17:24:00.4198 info    daemon/session/dns/SearchPaths : setting search paths ambassador default kube-node-lease kube-public kube-system test
2022-11-07 17:24:00.4199 info    daemon/session/dns/SearchPaths : Removing /etc/resolver/telepresence.tel2-search.local
```

Docker-run with fully qualified server name works:
```
telepresence intercept hello-server --namespace test \
  --docker-run -- --rm curlimages/curl:7.85.0 \
  -v --connect-timeout 10 \
  hello-server2.test:80
```
daemon.log:
```
2022-11-07 18:01:09.0591 info    daemon/session/dns/SearchPaths : setting search paths ambassador default kube-node-lease kube-public kube-system test test.svc.cluster.local. tel2-search
2022-11-07 18:01:09.0583 debug   daemon/session/watch-cluster-info : Opening tunnel for id tcp 10.60.0.0:55088 -> 10.60.0.40:33335
2022-11-07 18:01:09.0597 info    daemon/session/dns/SearchPaths : Generated new /etc/resolver/telepresence.tel2-search.local
2022-11-07 18:01:09.2917 debug   daemon/session/dns/Server : ServeDNS 44061 AAAA   hello-server.cluster.local.
2022-11-07 18:01:09.2918 debug   daemon/session/dns/Server : 44061 AAAA   hello-server.cluster.local. -> NXDOMAIN 
2022-11-07 18:01:09.3264 debug   daemon/session/dns/Server : ServeDNS 55850 AAAA   hello-server.test.svc.cluster.local.
2022-11-07 18:01:09.3267 debug   daemon/session/dns/Server : Lookup AAAA "hello-server.test.svc.cluster.local."
2022-11-07 18:01:09.3667 debug   daemon/session/dns/Server : 55850 AAAA   hello-server.test.svc.cluster.local. -> NOERROR EMPTY
2022-11-07 18:01:09.3733 debug   daemon/session/dns/Server : ServeDNS 24175 AAAA   hello-server.cluster.local.
2022-11-07 18:01:09.3736 debug   daemon/session/dns/Server : 24175 AAAA   hello-server.cluster.local. -> NXDOMAIN 
2022-11-07 18:01:12.4792 debug   daemon/session/dns/Server : ServeDNS 27819 AAAA   hello-server.test.svc.cluster.local.
2022-11-07 18:01:12.4793 debug   daemon/session/dns/Server : 27819 AAAA   hello-server.test.svc.cluster.local. -> NOERROR EMPTY
2022-11-07 18:01:12.4802 debug   daemon/session/dns/Server : ServeDNS 21692 AAAA   hello-server.cluster.local.
2022-11-07 18:01:12.4803 debug   daemon/session/dns/Server : 21692 AAAA   hello-server.cluster.local. -> NXDOMAIN 
2022-11-07 18:01:12.5306 debug   daemon/session/dns/Server : ServeDNS 20070 A      hello-server.test.svc.cluster.local.
2022-11-07 18:01:12.5309 debug   daemon/session/dns/Server : Lookup A "hello-server.test.svc.cluster.local."
2022-11-07 18:01:12.5780 debug   daemon/session/dns/Server : 20070 A      hello-server.test.svc.cluster.local. -> NOERROR [10.64.9.110]
2022-11-07 18:01:13.0784 debug   daemon/session/dns/Server : ServeDNS 51899 A      hello-server2.test.
2022-11-07 18:01:13.0786 debug   daemon/session/dns/Server : Lookup A "hello-server2.test."
2022-11-07 18:01:13.0883 debug   daemon/session/dns/Server : ServeDNS 30105 AAAA   hello-server2.test.
2022-11-07 18:01:13.0884 debug   daemon/session/dns/Server : Lookup AAAA "hello-server2.test."
2022-11-07 18:01:13.1418 debug   daemon/session/dns/Server : 51899 A      hello-server2.test. -> NOERROR [10.64.2.211]
2022-11-07 18:01:13.2165 debug   daemon/session/dns/Server : 30105 AAAA   hello-server2.test. -> NOERROR EMPTY
2022-11-07 18:01:13.2201 debug   daemon/session/dns/Server : ServeDNS 18690 SOA    hello-server2.test.
2022-11-07 18:01:13.2203 debug   daemon/session/dns/Server : Lookup SOA "hello-server2.test."
2022-11-07 18:01:14.2698 debug   daemon/session/dns/Server : ServeDNS 18690 SOA    hello-server2.test.
2022-11-07 18:01:14.2964 debug   daemon/session/dns/Server : 18690 SOA    hello-server2.test. -> NXDOMAIN 
2022-11-07 18:01:14.2964 debug   daemon/session/dns/Server : 18690 SOA    hello-server2.test. -> SERVFAIL rpc error: code = Unimplemented desc = unsupported DNS query type SOA
2022-11-07 18:01:14.3107 debug   daemon/session/watch-cluster-info : Opening tunnel for id tcp 10.64.0.0:55092 -> 10.64.2.211:80
2022-11-07 18:01:14.5735 debug   daemon/session/dns/SearchPaths : [ambassador default kube-node-lease kube-public kube-system test test.svc.cluster.local. tel2-search] -> [ambassador default kube-node-lease kube-public kube-system test]
2022-11-07 18:01:14.5735 info    daemon/session/dns/SearchPaths : setting search paths ambassador default kube-node-lease kube-public kube-system test
2022-11-07 18:01:14.5736 info    daemon/session/dns/SearchPaths : Removing /etc/resolver/telepresence.tel2-search.local
```

Interesting is that we see the line containing
```
SERVFAIL rpc error: code = Unimplemented desc = unsupported DNS query type SOA
```
in both docker-run cases. But only the unqualified name`hello-server2` fails. Fully qualified `hello-server2.test` works.

All tested with v2.8.5.

Also with v2.7.6 all here listed commands do work.

**Comment by thallgren** [comment]
I found the cause of this. A fix will be included in 2.9.0. You can test it already by downloading the release candidate from https://app.getambassador.io/download/tel2/darwin/amd64/2.9.0-rc.7/telepresence (replace amd64 with arm64 if you're on a M1).

**Comment by craftey** [comment]
Tested with v2.9.1. Works now. Thank you!
